src/main/java/de/tum/in/www1/artemis/repository/UserRepository.java

```diff
@@ -616,6 +619,30 @@ default User getUser() {
         return unwrapOptionalUser(user, currentUserLogin);
     }
 
+    /**
+     * Finds user id by login
+     *
+     * @param login the login of the user to search
+     * @return optional of the user id if it exists, empty otherwise
+     */
+    @Query("""
+            SELECT u.id
+            FROM User u
+            WHERE u.login = :login
+            """)
+    Optional<Long> findIdByLogin(@Param("login") String login);
```

Comment:
Why do we need to get the user ID by login? Shouldn't the client already provide the user ID?